### DISCUSSION TITLE
Add AtlasTexture support to Light2D

### DISCUSSION THREAD

**Discussion by CrayonChimp** [godotengine/godot-proposals#10487] [2D]
### Describe the project you are working on

A top-down 2D farming game.

### Describe the problem or limitation you are having in your project

Creating a separate texture for each light2d node would clutter my file system.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Like Sprite2D, merging related textures into one big sprite keeps the file system clean.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Just like Sprite2D is enough.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think this is very useful in larger projects if I want to create a fantastic night.

### Is there a reason why this should be core and not an add-on in the asset library?

Because it is a superset of the current version.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Reply by arkology** [reply]
The most straightforward implementations that could exist (and this is suggested as workaround in source code) - create internal `ImageTexture` and use it.
But it may be only suitable for lights with not-changing texture, as `get_image()` is slow.

**Comment by reimgrab** [comment]
It's not only about having less files. AtlasTexture support would be a first step to bring dynamic Light2D textures back like they used to work in Godot 3 (shameless plug #9855).
